crates/polars-ops/src/series/ops/unique.rs

```diff
@@ -25,6 +26,60 @@ where
     out.into_inner()
 }
 
+fn unique_counts_boolean_helper(ca: &BooleanChunked) -> IdxCa {
+    if ca.is_empty() {
+        return IdxCa::new(ca.name(), [] as [IdxSize; 0]);
+    }
+
+    let ca = ca.rechunk();
+    let arr = ca.downcast_iter().next().unwrap();
+
+    if let Some(validity) = arr.validity() {
+        let num_true = arr.values().set_bits();
+        let num_null = validity.unset_bits();
+        let num_false = arr.len() - num_true - num_null;
+
+        let (true_index, false_index, null_index) = find_first_true_false_null(
```

Comment:
This is a convenience function that will scan the entire array looking for true/false/null until it finds all three (or exhausts the array). Because we know in advance which are present, it might be faster to first check for the cases where we only true+null or only false+null and then do a linear scan until we find the indexes for these. Otherwise, do the 3-value search via `find_first_true_false_null`.

Reply:
Aha, thanks! I guess even if we e.g. only have true+null, we only have to check which is the first element and the other will be given? And similarly when we have all three, we only have to scan for the two first, and the third will be given.